cmd/packaging.go

```diff
@@ -0,0 +1,472 @@
+package cmd
+
+import (
+	"fmt"
+	"github.com/otiai10/copy"
+	"github.com/spf13/cobra"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"runtime"
+	"strings"
+)
+
+var desktopPackagingPath = filepath.Join(buildPath, "packaging")
+
+func init() {
+	initPackagingCmd.AddCommand(initLinuxSnapCmd)
+	initPackagingCmd.AddCommand(initLinuxDebCmd)
+	rootCmd.AddCommand(initPackagingCmd)
+}
+
+var initPackagingCmd = &cobra.Command{
+	Use:   "init-packaging",
+	Short: "Create configuration files for a packaging format",
+}
+
+var initLinuxSnapCmd = &cobra.Command{
+	Use:   "linux-snap",
+	Short: "Create configuration files for snap packaging",
+	Run: func(cmd *cobra.Command, args []string) {
+		projectName := assertInFlutterProject().Name
+		assertHoverInitialized()
+
+		initLinuxSnap(projectName)
+	},
+}
+
+var initLinuxDebCmd = &cobra.Command{
+	Use:   "linux-deb",
+	Short: "Create configuration files for deb packaging",
+	Run: func(cmd *cobra.Command, args []string) {
+		projectName := assertInFlutterProject().Name
+		assertHoverInitialized()
+
+		initLinuxDeb(projectName)
+	},
+}
+
+var linuxPackagingDependencies = []string{"libx11-6", "libxrandr2", "libxcursor1", "libxinerama1"}
+
+func packagingFormatPath(packagingFormat string) string {
+	directoryPath, err := filepath.Abs(filepath.Join(desktopPackagingPath, packagingFormat))
+	if err != nil {
+		fmt.Printf("hover: Failed to resolve absolute path for %s directory: %v\n", packagingFormat, err)
+		os.Exit(1)
+	}
+	return directoryPath
+}
+
+func createPackagingFormatDirectory(packagingFormat string) {
+	if _, err := os.Stat(packagingFormatPath(packagingFormat)); !os.IsNotExist(err) {
+		fmt.Printf("hover: A file or directory named `%s` already exists. Cannot continue packaging init for %s.\n", packagingFormat, packagingFormat)
+		os.Exit(1)
+	}
+	err := os.MkdirAll(packagingFormatPath(packagingFormat), 0775)
+	if err != nil {
+		fmt.Printf("hover: Failed to create %s directory %s: %v\n", packagingFormat, packagingFormatPath(packagingFormat), err)
+		os.Exit(1)
+	}
+}
+
+func assertPackagingFormatInitialized(packagingFormat string) {
+	if _, err := os.Stat(packagingFormatPath(packagingFormat)); os.IsNotExist(err) {
+		fmt.Printf("hover: %s is not initialized for packaging. Please run `hover init-packaging %s` first.\n", packagingFormat, packagingFormat)
+		os.Exit(1)
+	}
+}
+
+func assertCorrectOS(packagingFormat string) {
+	if runtime.GOOS != strings.Split(packagingFormat, "-")[0] {
+		fmt.Printf("hover: %s only works on %s\n", packagingFormat, strings.Split(packagingFormat, "-")[0])
+		os.Exit(1)
+	}
+}
+
+func escapedProjectName(projectName string) string {
```

Comment:
This doesn't escape, it just removes dashes and underscores. Name should reflect that.